contracts/utils/contracts/src/LibBytes.sol

```diff
@@ -179,7 +179,7 @@ library LibBytes {
             "FROM_LESS_THAN_TO_REQUIRED"
         );
         require(
-            to < b.length,
+            to <= b.length,
```

Comment:
Since `from <= to` and `to <= b.length` we can now encounter the scenario where `from == to == b.length`. We could add an extra assertion that `from < b.length` since it's intended to be a valid index, although looking at the implementations I don't think it should overwrite any unintended memory. Could still be worthwhile to include tests for this scenario though.

Reply:
This edge case does still modify memory with `sliceDestructive`, although maybe that is to be expected. It shouldn't matter for `slice`, but I'll add the necessary tests.